lib/child_process.js

```diff
@@ -20,15 +20,10 @@ exports.fork = function(modulePath /*, args, options*/) {
 
   // Get options and args arguments.
   var options, args, execArgv;
-  if (Array.isArray(arguments[1])) {
-    args = arguments[1];
-    options = util._extend({}, arguments[2]);
-  } else if (arguments[1] && typeof arguments[1] !== 'object') {
-    throw new TypeError('Incorrect value of args option');
-  } else {
-    args = [];
-    options = util._extend({}, arguments[1]);
-  }
+
+  var opts = normalizeForkArgs(arguments, options);
```

Comment:
I don't think we need a `normalizeForkArgs()`, as this code is specific to `fork()`. If the logic needs to be tightened up, it can be done here. `fork()` is one of the simpler cases, as we only need to validate an optional array and an optional object.